nemo/collections/asr/parts/utils/offline_clustering.py

```diff
@@ -506,20 +506,22 @@ def getMultiScaleCosAffinityMatrix(
             This function generates an affinity matrix that is obtained by calculating
             the weighted sum of the affinity matrices from the different scales.
     """
-    multiscale_weights = multiscale_weights.to(device)
-    score_mat_list, repeated_tensor_list = [], []
+    multiscale_weights = torch.squeeze(multiscale_weights,dim=0).to(device)
+    
+    repeated_tensor_list = []
     session_scale_mapping_list = get_argmin_mat(timestamps_in_scales)
     scale_list = list(range(len(timestamps_in_scales)))
-    for scale_idx in scale_list:
-        mapping_argmat = session_scale_mapping_list[scale_idx]
-        emb_t = embeddings_in_scales[scale_idx].half().to(device)
-        score_mat_torch = getCosAffinityMatrix(emb_t)
-        repeat_list = getRepeatedList(mapping_argmat, torch.tensor(score_mat_torch.shape[0])).to(device)
-        repeated_tensor_0 = torch.repeat_interleave(score_mat_torch, repeats=repeat_list, dim=0)
-        repeated_tensor_1 = torch.repeat_interleave(repeated_tensor_0, repeats=repeat_list, dim=1)
-        repeated_tensor_list.append(repeated_tensor_1)
-    repp = torch.stack(repeated_tensor_list).float()
-    fused_sim_d = torch.matmul(repp.permute(2, 1, 0), multiscale_weights.t()).squeeze(2).t()
+    fused_sim_d = torch.zeros(len(timestamps_in_scales[-1]),len(timestamps_in_scales[-1]))
+
+    for scale_idx in scale_list:       
+        with torch.no_grad():            
+            mapping_argmat = session_scale_mapping_list[scale_idx]
+            emb_t = embeddings_in_scales[scale_idx].half().to(device)
+            score_mat_torch = getCosAffinityMatrix(emb_t)
+            repeat_list = getRepeatedList(mapping_argmat, torch.tensor(score_mat_torch.shape[0])).to(device)
+            repeated_tensor_0 = torch.repeat_interleave(score_mat_torch, repeats=repeat_list, dim=0)            
+            repeated_tensor_1 = torch.repeat_interleave(repeated_tensor_0, repeats=repeat_list, dim=1)
+            fused_sim_d = fused_sim_d + multiscale_weights[scale_idx] * repeated_tensor_1
```

Comment:
Thank you for accepting my suggestion and pushing this.
Wouldn't it be better to clear memory here by doing the following:

```python
del emb_t, repeated_tensor_0, repeated_tensor_1
torch.empty_cache()
```
I need to check whether this helps or not.

Reply:
This doesn't work in torch.jit.script decorator. I have added torch.empty.cache() outside the clustering code and now it is clearning memory.